doc/contribute/nest_with_ides.rst

```diff
@@ -0,0 +1,103 @@
+Developing NEST with IDEs
+=========================
+
+Integrated development environments can make coding more efficient and fun.
+Here are some recipes based on practical experience on how to use
+IDEs in NEST development. The information is not meant to be complete,
+cross-checked or up-to-date but is provided on a best-effort basis to get
+you started. Kindly contribute your experiences.
+
+Since we generally recommend out-of-source builds, only such are discussed here.
+
+.. contents:: We currently have recipes for the following IDEs
+   :local:
+   :depth: 2
+
+Eclipse
+-------
+
+This recipe describes how to set up Eclipse for editing, building and
+running NEST. The description here was tested on macOS 11.
+
+Requirements and limitations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Focus on single build configuration
+* Assumes all dependencies (OpenMPI, GSL, etc) installed in a Conda environment
+* Does not support debugging on macOS (because Eclipse does not support lldb)
+* Does not read the NEST `.clang-format` file, so code formatting may
+  be incorrect
+* I experimented with better CMake support using ``cmake4eclipse``,
+  but did not figure it out. The instructions below rely on running
```

Comment:
In general, it would be good to avoid the pronoun "I" (@jessica-mitchell and I are currently discussing the documentation style guide). We can also reword the sentence to use positive language. 

How about "We are in the process of experimenting with better CMake support using ``cmake4eclipse``. The instructions below..."?

Reply:
@sarakonradi I reworded the point, but we are not in the process of experimenting with cmake4eclipse, absolutely no time for that ;).